lisa/utils.py

```diff
@@ -3845,4 +3845,34 @@ def __reduce__(self):
         return (self._make_instance, (self._ctor, dct))
 
 
+class LazyMapping(Mapping):
+    """
+    Lazy Mapping dict-like class for elements evaluated on the fly.
+
+    It takes the same set of arguments as a dict with keys as the mapping keys
+    and values as  closures that take a key and return the value. The class does
+    no automatic memoization but memoization can easily be achieved using something
+    like lru_cache from functools, as shown in the example below.
+
+    **Example**::
+
+        LazyMapping({
+            f"{job.label}-{job.iteration}": lru_cache()(lambda k: Trace(job.get_artifact_path('trace-cmd-bin'), **self._trace_kwargs))
```

Comment:
`**self._trace_kwargs` must be removed, this is an private attribute so we can definitely not have examples of a public API showing it.

The example can simply show how to compute `key + 42` without requiring the user to know anything about WA